RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -103,15 +103,15 @@ DAClusterizerInZT_vect::DAClusterizerInZT_vect(const edm::ParameterSet& conf) {
 namespace {
   inline double local_exp(double const& inp) { return vdt::fast_exp(inp); }
 
-  inline void local_exp_list(double const* arg_inp, double* arg_out, const unsigned arg_arr_size) {
-    for (unsigned i = 0; i != arg_arr_size; ++i)
-      arg_out[i] = vdt::fast_exp(arg_inp[i]);
+  inline void local_exp_v(double const* arg_inp, double* arg_out, const unsigned arg_arr_size) {
```

Comment:
addd restrict?
btw linking to vdt library may have undesired side effects as it override also scalar std functions.
Please also note that vdt can also be compiled as a fat library with runtime switch to use best vector-width

Reply:
>  btw linking to vdt library may have undesired side effects as it overrides also scalar std functions.

what would be the side-effects?
The last set of tests showed no differences in the results.

Reply:
overwrites sin, cos, etc

Reply:
got it.
From the tests, however, it looks like at least on intel (supposedly where jenkins runs the tests) the results are numerically the same.

Reply:
some trigonometric functions (sin, cos and atan2) are used in DAClusterizerInZT_vect, but I think the same thing could actually be done without them if this helps.